### PR TITLE
Feature/#171 context handling

### LINKED ISSUE #171: Better handling of context fields.
### Description

The following modifications need to be made to the `@context` field in order to provide better support for custom credential usage:

1) Allow for inline context definition.
2) Change all Jolocom context definitions to be inline.
3) Provide a simple `defaultContext` that contains all credential specific terms.
4) Inject custom contexts into the `defaultContext`

### TODO
- [ ] Implement.

### PR DESCRIPTION
**Pull request by Exulansis**
Fixes #171 .

### CODE REVIEW

**Comment by nataschaberg** [comment]
Failing tests.

File: ts/credentials/credential/credential.ts [hunk 1]
```diff
@@ -1,48 +1,42 @@
...
-import { IClaimAttrs, ICredentialAttrs, ICredentialCreateAttrs } from './types'
+import { ICredentialAttrs, IClaimSection } from './types'
+import { BaseMetadata, validContextEntry } from 'cred-types-jolocom-core'
+import { defaultContext } from '../../utils/contexts'
 
 @Exclude()
 export class Credential {
   @Expose()
-  private '@context': string[] | object[]
+  private '@context': Array<validContextEntry>
```
[the remark this answers is not in the document]
Comment (Exulansis) [reply ts/credentials/credential/credential.ts]: Agree

File: tests/data/credential/credential.ts [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
-  id: 'did:jolo:test',
   givenName: 'Eugeniu',
   familyName: 'Rusu'
 }
 
-export const customClaimMetadata = {
-  context: ['http://test.com'],
-  fieldNames: ['test', 'secondTest'],
-  optionalFieldNames: [],
+export interface ClaimInterface extends BaseMetadata {
```
[the remark this answers is not in the document]
Comment (Exulansis) [reply tests/data/credential/credential.ts]: Agree

File: ts/credentials/credential/credential.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (charleenfei) [reply ts/credentials/credential/credential.ts]: this issue will be resolved on the cred-types-jolocom-core repo.

File: tests/data/credential/credential.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (charleenfei) [reply tests/data/credential/credential.ts]: renamed

**Comment by katszwn** [comment]
- [ ] rename keyTypes to be upper case

**Pull request merged**